frontend/cypress/integration/featureFiles/services.feature

```diff
@@ -60,3 +60,34 @@ Feature: Kiali Services page
     And user filters for label "app=productpage"
     Then user sees "productpage" in the table
     And table length should be 1
+
+  @services-page
+  Scenario: The healthy status of a service is reported in the list of services
+    Given a service in the cluster with a healthy amount of traffic
+    When I fetch the list of services
```

Comment:
You don't need to include this check here since the Background already navigates to the services page with the check `And user is at the "services" page`. Moreover, the user isn't really fetching the list of services themselves, the app is doing that. The user is just navigating to a particular page and expecting to see a list of services show up after the page loads.

Reply:
> You don't need to include this check here since the Background already navigates to the services page with the check `And user is at the "services" page`.

I'm not thinking in \<the services page\> as initial context, but more as an action that the user would want to do :confused: 

> The user is just navigating to a particular page and expecting to see a list of services show up after the page loads.

That would be entering in implementation details, no? :confused: 

I'm trying to follow what is mentioned in this small page: https://cucumber.io/docs/bdd/examples/

I know it is very hard to assume no computers in our case. But we can make the assumption that Kiali does not exist and you still need to "see" the healthy status of a service. Perhaps, somebody will need to prepare a list on paper and would place colored sticky markers reflecting health status of each item in the list...

Well... I'll try to see if there is some way to fix what you are commenting here.

Reply:
> the Background already navigates to the services page

BTW, I'm writing a few new tests for the Istio Config page. In these tests I need to apply changes to the env (similar to the sidecar tests).

There is the common pattern that the `Background` is pre-navigating to the Istio Config page. However, since env changes come after the `Background`, I anyway need to refresh the page via a `When the user fetches the list of Istio CRDs` sentence, similarly to how I was doing here. So, that makes me doubt if the `Background` is really a good place to do pre-navigation if you may need to do more preparation in `Given`s on each scenario.

Reply:
I just created #5085, a small draft PR of the other tests I'm starting to write, to let you see the need for reloading the page even if the same page is already loaded in the `Background`.

It's just to reach an agreement about how it should be done and use the same pattern here and there, as I think all tests should be consistent.

Reply:
> I'm not thinking in <the services page> as initial context, but more as an action that the user would want to do confused
> 
> > The user is just navigating to a particular page and expecting to see a list of services show up after the page loads.
> 
> That would be entering in implementation details, no? confused

That's fair the wording could probably be updated to be something less implementation specific like `Given I am viewing my "services"` or `Given I am viewing my "bookinfo/details" "service"` but, for me, "fetching" doesn't really describe what the user is doing here. They're not opening up a network request to go get each service or as in your sticky note example writing out each note and arranging it themselves, rather they are just viewing all their services. Maybe this isn't strictly adhering to the Gherkin syntax but I also think it's more understandable to describe the tests in terms of what we're actually testing i.e. we're testing the services list page.

> There is the common pattern that the Background is pre-navigating to the Istio Config page. However, since env changes come after the Background, I anyway need to refresh the page via a When the user fetches the list of Istio CRDs sentence, similarly to how I was doing here. So, that makes me doubt if the Background is really a good place to do pre-navigation if you may need to do more preparation in Givens on each scenario.

Couldn't you also hit the refresh button after patching your env? That's what a user would do to have Kiali pickup new changes to their env when they've paused the autorefresh like we do in the tests.

From the docs on the Background section:
> Occasionally you’ll find yourself repeating the same Given steps in all of the scenarios in a Feature.
>
> Since it is repeated in every scenario, this is an indication that those steps are not essential to describe the scenarios; they are incidental details. You can literally move such Given steps to the background, by grouping them under a Background section.

The fact that we're on the services page is an incidental detail here since we have to be on the services page in order to test the services page. Every test otherwise would have that `Given` step. If the page loading doesn't belong in the `Background` section, I'm not really sure what does.

Reply:
> they are just viewing all their services.

Well, I could use `When the user looks at the services list`, if that sounds better. The implementation would remain, though.

> Maybe this isn't strictly adhering to the Gherkin syntax but I also think it's more understandable to describe the tests in terms of what we're actually testing 

I agree. But IMHO that route is not aligned with cucumber vision.

> Couldn't you also hit the refresh button after patching your env?

That's an option, I think.

> From the docs on the Background section:

Well, I agree with the docs. But `Background` is limited to `Given` steps, while this is a `When`.

Not sure, but I'm so-so reading that you want this `When` converted to a `Given`. I know that in typescript all these things are aliases. But, conceptually they are different. So, I'm not sure that we should be changing/using the terms indistinctly.

> The fact that we're on the services page is an incidental detail here since we have to be on the services page in order to test the services page. Every test otherwise would have that Given step. If the page loading doesn't belong in the Background section, I'm not really sure what does.

IMHO, that is too focused on implementation. And about the background, well, the only thing I see valid without entering in implementation details is the sentence telling that the logged in user is an administrator. Change that background to a non-admin user and you can then start thinking in very different scenarios.

All that said... well... I'm not leading this effort. So, meh, I'll rework this PR to your liking.

Reply:
> Not sure, but I'm so-so reading that you want this When converted to a Given. I know that in typescript all these things are aliases. But, conceptually they are different. So, I'm not sure that we should be changing/using the terms indistinctly.

Yeah `When` statements don't seem to make much sense in `Background`. Maybe something like `Given user is looking at their services list`?

> All that said... well... I'm not leading this effort. So, meh, I'll rework this PR to your liking.

It's a group effort and you've spent much more time reading the cucumber docs/recommendations. I appreciate your input and perspective on how we're implementing these tests. Adherence to cucumber has not been as much of a priority as getting coverage and encouraging folks to write tests. Ultimately whichever syntax makes the tests easier to read, write, and maintain for contributors is the way I prefer.

Reply:
> It makes me wonder why we chose it, if perhaps something else would have been a better fit (like the regular specs).

That's a good question. I believe it was primarily chosen so that the testing requirements for a feature could be formally defined separately from the feature itself. QE could write/review the feature file independently of the implementation. I think it still serves that purpose even if we're not following the recommended syntax exactly.